workers/src/main/scala/workers/AgsServer.scala

```diff
@@ -20,18 +25,34 @@ object AgsServer extends WorkerServer[IO, AgsMessage.Request] {
   @JSExport
   def runWorker(): Unit = run.unsafeRunAndForget()
 
-  protected val handler: Logger[IO] ?=> IO[Invocation => IO[Unit]] = IO { invocation =>
-    invocation.data match {
-      case AgsMessage.Request(id, constraints, wavelength, base, basePos, params, candidates) =>
-        Logger[IO].debug(s"AGS request for $id") >>
-          IO.delay(
-            Ags
-              .agsAnalysis(constraints, wavelength, base, basePos, params, candidates)
-              .sorted(AgsAnalysis.rankingOrdering)
-          ).flatMap(r =>
-            Logger[IO].debug(s"AGS response for $id") >>
-              invocation.respond(r)
-          )
-    }
-  }
+  private val AgsCacheVersion: Int = 1
+
+  private val CacheRetention: Duration = Duration.ofDays(60)
+
+  def agsCalculation(r: AgsMessage.Request): IO[List[AgsAnalysis]] =
+    IO.delay(
+      Ags
+        .agsAnalysis(r.constraints,
+                     r.wavelength,
+                     r.baseCoordinates,
+                     r.position,
+                     r.params,
+                     r.candidates
+        )
+        .sorted(AgsAnalysis.rankingOrdering)
+    )
+
+  protected val handler: Logger[IO] ?=> IO[Invocation => IO[Unit]] =
+    for
+      self  <- IO(dom.DedicatedWorkerGlobalScope.self)
+      cache <- Cache.withIDB[IO](self.indexedDB.toOption, "ags")
+      _     <- cache.evict(CacheRetention).start
+    yield invocation =>
+      invocation.data match {
+        case req @ AgsMessage.Request(_, _, _, _, _, _, _) =>
+          val cacheableRequest =
+            Cacheable(CacheName("ags"), CacheVersion(AgsCacheVersion), agsCalculation)
+          (IO.now() >>= (now => cache.evict(now.minus(CacheRetention)).start)) >>
```

Comment:
Do we want to run eviction on every request? We are alrady running it on startup.

Reply:
ah you're right. I guess it will be kind of expensive if it gets longer